.bash_profile

```diff
@@ -35,8 +35,8 @@ elif [ -f /etc/bash_completion ]; then
 fi;
 
 # Enable tab completion for `g` by marking it as an alias for `git`
-if type _git &> /dev/null; then
-	complete -o default -o nospace -F _git g;
+if type __git_wrap__git_main &> /dev/null; then
+    complete -o default -o nospace -F __git_wrap__git_main g
```

Comment:
https://github.com/git/git/commit/5a067ba9d04eebc92ad77f101b785219238f4f1e might be of interest here. Unless I'm misunderstanding, it sounds like we can do

```
__git_complete g git
```

to avoid reaching into "private" details like `__git_wrap__git_main`?

Reply:
Thanks @lucthev, that's indeed the way to make use of git's autocompletion, confirmed in the documentation 🤓
https://github.com/git/git/blob/master/contrib/completion/git-completion.bash#L39